lib/influx.rb

```diff
@@ -182,11 +182,23 @@ def noise_candidates(start_date = nil, end_date = nil)
       }.compact
     end
 
-    def save_categories(data)
+    def save_categories(opts)
+      return if opts[:data].empty?
       timeseries = @config['series']
-      data.each do |incident, category|
-        query = "select * from #{timeseries} where id = '#{incident}'"
-        current_point = @influxdb.query(query)[timeseries].first
+      oldest =  Chronic.parse(opts[:start_date], :guess => false).first.to_i
+      newest =  Chronic.parse(opts[:end_date], :guess => false).last.to_i
+      begin
+        entries = @influxdb.query "select * from #{timeseries} where time > #{oldest}s and time < #{newest}s"
+        entries = entries.empty? ? [] : entries[timeseries]
+      rescue InfluxDB::Error => e
+        if e.message.match(/^Couldn't find series/)
```

Comment:
> However, if we do get here, should we still try writing the data on the point we have, or completely aborting? My code currently does the former, and your edit is suggesting the latter.

Yeah, it was a suggestion on my part and at the time of writing I thought the 2 bits of code were equivalent but they're not, oops. I agree trying to write is better than completely aborting.